pkg/plugin/builtin/v1/sandbox.go

```diff
@@ -28,27 +36,71 @@ func runSandbox[REQ any, RES any](
 	f func(context.Context, REQ) (RES, error),
 	ctx context.Context, // context is the second parameter on purpose
 	req REQ,
+	logger log.CtxLogger,
 ) (res RES, err error) {
-	defer func() {
-		if r := recover(); r != nil {
-			panicErr, ok := r.(error)
-			if !ok {
-				panicErr = cerrors.Errorf("panic: %v", r)
-			}
-			err = panicErr
+	c := sandboxChanPool.Get().(chan any)
+
+	returnResponse := func(ctx context.Context, res RES, err error) {
+		defer sandboxChanPool.Put(c)
+		select {
+		case <-ctx.Done():
+			// The context was cancelled, nobody will fetch the result.
+			logger.Error(ctx).
+				Any("response", res).
+				Err(err).
+				Msg("context cancelled when trying to return response from builtin connector plugin (this message comes from a detached plugin)")
+		case c <- res:
+			// The result was sent, now send the error if any and return the
+			// channel to the pool.
+			c <- err
 		}
+	}
+
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				err, ok := r.(error)
+				if !ok {
+					err = cerrors.Errorf("panic: %v", r)
```

Comment:
Out of curiosity, when is this possible? If a function panicked, then recover() should always return the error, or no?

Reply:
So `recover()` will return whatever value was passed to `panic()`, which can be anything. Here's an example: https://go.dev/play/p/XC48MpRhbFY

Reply:
Thanks for the example!